### PR TITLE
fix(content-block): render optional linklist in layout

### PR DESCRIPTION
**Pull request by annawen1**
### Related Ticket(s)

Linklist is rendering too high on a pattern #1920

### Description

wrap `ContentBlock` content and heading in `Layout` component to ensure the `LinkList` aligns properly with the copy.

Screen Shot 2020-04-10 at 12 15 46 PM

### Changelog

**Changed**

- wrap the `heading` and `content` in `Layout` component if `LinkList` is passed in

### CODE REVIEW

**Comment by jeffchew** [comment]
@annawen1 can the stories for the content block decorator patterns be added to include the LinkList to optionally turn on or off? @photodow which patterns would need to have this as an example?

cc: @kennylam

**Comment by kennylam** [comment]
@jeffchew @photodow Should we leave that right column open to accept any component and not just `LinkList`? So instead of taking `LinkList` as a prop it would just be generic child content for that right col?

(This looks great btw, @annawen1)

**Comment by photodow** [comment]
> @jeffchew @photodow Should we leave that right column open to accept any component and not just LinkList?

I believe so. Other things will eventually need to live in the right hand side of the content block.

File: packages/react/src/patterns/sub-patterns/ContentBlock/ContentBlock.js [hunk 1]
```diff
@@ -55,10 +56,58 @@ const ContentBlock = ({ heading, copy, children, customClassName, cta }) => {
...
+              <div></div>
+            </>
+          )
+        : title}
+      {linkList
+        ? _layoutWrap(
+            <>
+              <div>{content}</div>
+              <div>
+                <LinkList {...linkList} />
```
Comment (photodow) [inline packages/react/src/patterns/sub-patterns/ContentBlock/ContentBlock.js:91]: Should link list be built directly into content block, or should it just allow an adopter to pass in link list and other items?

File: packages/react/src/patterns/sub-patterns/ContentBlock/ContentBlock.js [hunk 2]
```diff
@@ -55,10 +56,58 @@ const ContentBlock = ({ heading, copy, children, customClassName, cta }) => {
...
 
+/**
+ * wraps content in layout component
+ *
+ * @private
+ * @param {Element} content content elements
+ * @returns {*} jsx cta component
+ */
+const _layoutWrap = content => (
+  <Layout type="2-1">{content.props.children}</Layout>
```
Comment (photodow) [inline packages/react/src/patterns/sub-patterns/ContentBlock/ContentBlock.js:108]: I believe the nested option would need to be passed in to make this work. Probably should also allow for other props like border to be turned on as well.

File: packages/react/src/patterns/sub-patterns/ContentBlock/ContentBlock.js (hunk 1 above)
Comment (annawen1) [reply packages/react/src/patterns/sub-patterns/ContentBlock/ContentBlock.js:91]: ok, let me try that out

File: packages/react/src/patterns/sub-patterns/ContentBlock/ContentBlock.js (hunk 2 above)
Comment (annawen1) [reply packages/react/src/patterns/sub-patterns/ContentBlock/ContentBlock.js]: done

**Comment by annawen1** [comment]
me trying to update my branch + lag = accidentally merging this PR in to master...have a PR to revert the changes if necessary...

**Comment by jeffchew** [comment]
> me trying to update my branch + lag = accidentally merging this PR in to master...have a PR to revert the changes if necessary...

@annawen1 I'll take a look, but one thing I noticed, there's no documentation on how to do the new `aside` feature. I also mentioned before that we should add this to the storybook stories for the decorator patterns to show how it works in those contexts.

**Pull request merged**